api/spec/controllers/mno_enterprise/jpi/v1/marketplace_controller_spec.rb

```diff
@@ -111,5 +114,28 @@ def hash_for_apps(apps)
         expect(JSON.parse(response.body)).to eq(JSON.parse(hash_for_app(app).to_json))
       end
     end
+     
+    describe 'GET #index' do
+      subject { get :index }
+
+      context 'when multiples apps' do 
+        let!(:app2) { build(:app, rank: 0 ) }
```

Comment:
You can use `let` as you're referencing `app1` and `app2` in your before block.

You should test with a `nil` rank as well as this is a special case.